synfig-core/src/tool/joblistprocessor.cpp

```diff
@@ -213,9 +214,10 @@ bool setup_job(Job& job, const TargetParam& target_parameters)
 		}
 	}
 
-	// Set the threads for the target
+	// Set the threads and render engine for the target
 	if (job.target && Target_Scanline::Handle::cast_dynamic(job.target))
 		Target_Scanline::Handle::cast_dynamic(job.target)->set_threads(SynfigToolGeneralOptions::instance()->get_threads());
+                Target_Scanline::Handle::cast_dynamic(job.target)->set_engine(job.render_engine);
```

Comment:
Three equal `dynamic_cast`  in a row?
I see the original code has two, but assign it to a variable and reuse the dynamic cast result.

By the way, the original code should handle both `Target_Scanline` and `Target_Tile` to `set_threads()` and `set_engine()`.

Reply:
Are you missing curly brackets `{}` to enclose the `if` contents?

Reply:
dynamic_cast is slow. So we should avoid call it.
Instead of three equal calls, do it only once:

```c++
if (job.target) {
	Target_Scanline::Handle scanline_target = Target_Scanline::Handle::cast_dynamic(job.target);
	scanline_target->set_threads(SynfigToolGeneralOptions::instance()->get_threads());
	scanline_target->set_engine(job.render_engine);
}
```

Reply:
So should I dynamic_cast it to Target_Tile also and then set the threads and engine 🤔 ?

Reply:
Yes, you should.

Just fixing my previous code:
```c++
if (job.target) {
	if (auto scanline_target = Target_Scanline::Handle::cast_dynamic(job.target)) {
		scanline_target->set_threads(SynfigToolGeneralOptions::instance()->get_threads());
		scanline_target->set_engine(job.render_engine);
	}
}
```